ipydatagrid/datagrid.py

```diff
@@ -364,12 +359,23 @@ def data(self, dataframe):
 
         schema["primaryKeyUuid"] = guid_key
 
-        self._data = {
+
+        return {
             "data": data,
             "schema": schema,
             "fields": [{field["name"]: None} for field in schema["fields"]],
         }
 
+    @data.setter
+    def data(self, dataframe):
+        # Reference for the original frame column and index names
+        # This is used to when returning the view data model
+        self.__dataframe_reference_index_names = dataframe.index.names
+        self.__dataframe_reference_columns = dataframe.columns
+        dataframe = dataframe.copy()
+
+        self._data = DataGrid.generate_data_object(dataframe, "ipydguuid")
```

Comment:
any reason we're not using just self here?

Reply:
static method

Reply:
you can and should call a static method via self